ui/src/pages/report/InputDialog.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useState } from "react";
+import Dialog from "@material-ui/core/Dialog";
+import DialogTitle from "@material-ui/core/DialogTitle";
+import DialogContent from "@material-ui/core/DialogContent";
+import DialogActions from "@material-ui/core/DialogActions";
+import Button from "@material-ui/core/Button";
+import TextField from "@material-ui/core/TextField";
+import Select from "@material-ui/core/Select";
+import InputLabel from "@material-ui/core/InputLabel";
+import FormControl from "@material-ui/core/FormControl";
+import MenuItem from "@material-ui/core/MenuItem";
+
+export interface Selection {
+  items : string[]
+}
+
+export interface Field {
+  name : string
+  label : string
+  type : string | Selection
+}
+
+export interface InputDialogProps {
+  open : boolean
+  title : string
+  fields : Field[]
+  onClose : (state : any) => Promise<void>
+}
+
+export const InputDialog = (props : InputDialogProps) => {
+  const [ state, setState ] = useState<any>({});
+
+  return (
+    <Dialog open={props.open} onClose={() => props.onClose(undefined)} maxWidth="sm" fullWidth>
+      <DialogTitle>
+        {props.title}
+      </DialogTitle>
+      <DialogContent>
+        {props.fields.map((f, i) => {
+          if ((f.type as Selection).items) {
```

Comment:
Oh cool, I didn't realize that you can do that! That's an elegant way to check, but is a shame that the Typescript compiler can't reuse that 4 lines below.